scripts/pi-hole/js/db_graph.js

```diff
@@ -63,32 +63,84 @@ function compareNumbers(a, b) {
   return a - b;
 }
 
+function computeInterval(from, until) {
+  // Compute interval to obtain about 200 values
+  var num = 200;
+  // humanly understandable intervals (in seconds)
+  var intervals = [
+    10,
+    20,
+    30,
+    60,
+    120,
+    180,
+    300,
+    600,
+    900,
+    1200,
+    1800,
+    3600,
+    3600 * 2,
+    3600 * 3,
+    3600 * 4,
+    3600 * 6,
+    3600 * 8,
+    3600 * 12,
+    3600 * 24,
+    3600 * 24 * 7,
+    3600 * 24 * 30,
+  ];
+
+  var duration = until - from;
+  if (duration / (num * intervals[0]) < 1) {
+    return intervals[0];
+  }
+
+  var preverr = Number.MAX_VALUE,
+    err;
+  for (var i in intervals) {
+    if (Object.prototype.hasOwnProperty.call(intervals, i)) {
+      err = Math.abs(1 - duration / (num * intervals[i]));
+      // pick the interval with least deviation
+      // from selected duration
+      if (preverr < err) {
+        // console.log("interval=", intervals[i - 1], "duration=", duration, "values=", duration / intervals[i - 1]);
```

Comment:
Do we need this comment still here?

Reply:
I don't think so. It was useful during the tests, but can be removed now.

Reply:
Removed this comment in the latest commit.